contracts/base/escrow/dapps/YVault.sol

```diff
@@ -0,0 +1,144 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+// Contracts
+import "../../BaseEscrowDapp.sol";
+
+// External Libraries
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/Address.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
+
+// Interfaces
+import "./YVaultInterface.sol";
+import "../../../providers/yearn/IVault.sol";
+
+/*****************************************************************************************************/
+/**                                             WARNING                                             **/
+/**                      DAPP CONTRACT IS AN EXTENSION OF THE ESCROW CONTRACT                       **/
+/**  ---------------------------------------------------------------------------------------------  **/
+/**  Because there are multiple dApp contracts, and they all extend the Escrow contract that is     **/
+/**  itself upgradeable, they cannot have their own storage variables as they would cause the the   **/
+/**  storage slots to be overwritten on the Escrow proxy contract!                                  **/
+/**                                                                                                 **/
+/**  Visit https://docs.openzeppelin.com/upgrades/2.6/proxies#upgrading-via-the-proxy-pattern for   **/
+/**  more information.                                                                              **/
+/*****************************************************************************************************/
+/**
+    @notice This contract is used to define the yVault dApp actions available. All dapp actions are invoked via
+        delegatecalls from Escrow contract, so this contract's state is really Escrow.
+    @author develop@teller.finance
+ */
+contract YVault is YVaultInterface, BaseEscrowDapp {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    
+
+    /** External Functions */
+    /**
+        @notice Deposits the specified amount of the native unwrapped token (same as token() returns) into the Vault
+        @param tokenAddress The address of the token being deposited
+        @param amount The amount of tokens to be deposited into the vault
+     */
+    function deposit(
+        address tokenAddress,
+        uint256 amount
+    )
+        public
+        onlyOwner()
+    {
+        IVault iVault = _getYVault(tokenAddress);
```

Comment:
Do we need to validate whether the returned instance is not an address(0x0)?

Reply:
Doing this inside of the AssetSettings contract when it's being set